### ISSUE TITLE
torch.load(weights_only=True) fails with unclear error for .pth files containing non-tensor metadata

### ISSUE THREAD

**Issue by oeway** [bioimage-io/core-bioimage-io-python#484]
## Problem

When loading PyTorch weights with `bioimageio.core`, the call in `pytorch_backend.py:157` uses:

```python
state = torch.load(f, map_location=devices[0], weights_only=True)
```

If a `.pth` file contains non-tensor objects (e.g., numpy arrays for normalization stats, as is common with deepflash2, MONAI, and other frameworks that bundle extra metadata in the checkpoint), this raises a PyTorch `WeightsUnpicklingError` or `UnpicklingError` with a message that doesn't explain the bioimageio-specific fix.

## Steps to reproduce

1. Take any `.pth` checkpoint that stores metadata alongside the state dict (e.g. deepflash2's format: `{'model': state_dict, 'stats': np.array(...)}`)
2. Reference it in a `bioimageio.yaml` under `weights.pytorch_state_dict.source`
3. Run `bioimageio test rdf.yaml`

## Expected behavior

A clear error message such as:
> "The weights file contains non-tensor objects (e.g. numpy arrays). bioimageio requires a pure state dict. Save only the model parameters: `torch.save(checkpoint['model'], 'weights.pt')`"

Or alternatively: gracefully retry with `weights_only=False` after warning the user.

## Actual behavior

Raw PyTorch error propagates without actionable guidance for the user.

## Fix suggestion

In `pytorch_backend.py`, wrap the `torch.load` call:

```python
try:
    state = torch.load(f, map_location=devices[0], weights_only=True)
except Exception as e:
    raise ValueError(
        f"Failed to load weights with weights_only=True: {e}\n"
        "If your .pth file contains non-tensor metadata (numpy arrays, strings, etc.), "
        "extract only the state dict before packaging:\n"
        "  data = torch.load('original.pth', weights_only=False)\n"
        "  torch.save(data['model_key'], 'weights.pt')"
    ) from e
```

## Context

Discovered during automated model submission testing using the bioimageio-models agent skill.
Model: matjesg/cFOS_in_HC (deepflash2, HuggingFace)
bioimageio.core version tested: latest from PyPI

/cc @fynnbe
